velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -324,5 +325,45 @@ TEST_F(Re2FunctionsTest, regexExtractConstantPatternNoGroupId) {
   EXPECT_EQ(extract("a b245 c3", "\\d+"), "245");
 }
 
+TEST_F(Re2FunctionsTest, likePattern) {
+  auto like =
+      ([&](std::optional<std::string> str, std::optional<std::string> pattern) {
+        return evaluateOnce<std::string>("like(c0, '" + *pattern + "')", str);
+      });
+
+  EXPECT_EQ(like("abc", "%b%"), "abc");
+  EXPECT_EQ(like("bcd", "%b%"), "bcd");
+  EXPECT_EQ(like("cde", "%b%"), std::nullopt);
+
+  EXPECT_EQ(like("abc", "_b%"), "abc");
+  EXPECT_EQ(like("bcd", "_b%"), std::nullopt);
+  EXPECT_EQ(like("cde", "_b%"), std::nullopt);
+
+  EXPECT_EQ(like("abc", "b%"), std::nullopt);
```

Comment:
Would you benchmark prefix patterns like this to see if using re2 is any slower then doing basic strcmp?